### ISSUE TITLE
Still prompting to daemon-reload after doing so

### ISSUE THREAD

**Issue by amo13** [systemd/systemd#17312]
**systemd version the issue has been seen with**
246.6

**Used distribution**
archlinux

**Linux kernel version used** (`uname -a`)
Linux server 5.8.13-arch1-1

**CPU architecture issue was seen on**
x86_64

**Unexpected behaviour you saw**
I try to use a drop-in configuration to receive a notification about the results of my btrfs-scrub@.service. Before adding the drop-in, the unit looks like this:
```
$ sudo systemctl cat btrfs-scrub@
# /usr/lib/systemd/system/btrfs-scrub@.service
[Unit]
Description=Btrfs scrub on %f

[Service]
Nice=19
IOSchedulingClass=idle
KillSignal=SIGINT
ExecStart=/usr/bin/btrfs scrub start -B %f

# /etc/systemd/system/service.d/toplevel-override.conf
[Unit]
OnFailure=failure-notification@%n
```
With `sudo systemctl edit btrfs-scrub@` I create a drop-in config so as to have
```
$ sudo systemctl cat btrfs-scrub@
# /usr/lib/systemd/system/btrfs-scrub@.service
[Unit]
Description=Btrfs scrub on %f

[Service]
Nice=19
IOSchedulingClass=idle
KillSignal=SIGINT
ExecStart=/usr/bin/btrfs scrub start -B %f

# /etc/systemd/system/btrfs-scrub@.service.d/override.conf
[Service]
ExecStopPost=/bin/sh -c 'journalctl -u %n -S -12h --no-pager | gotify-cli push'

# /etc/systemd/system/service.d/toplevel-override.conf
[Unit]
OnFailure=failure-notification@%n
```
Now even after having done another `systemctl daemon-reload`, I am told that the config file changed on disk and that a reload is necessary:
```
$ sudo systemctl start btrfs-scrub@-
Warning: The unit file, source configuration file or drop-ins of btrfs-scrub@-.service changed on disk. Run 'systemctl daemon-reload' to reload units.
```
This message appears misleading to me because the service runs fine and I do receive the notification. 

**Steps to reproduce the problem**
* add a toplevel drop-in as indicated above
* add a drop-in as indicated above
* start the service

**Comment by poettering** [comment]
Pretty sure this is already fixed by #16885, which is post v246. Closing.

**Comment by amo13** [comment]
The issue seems to be back. I am on v247.3 on archlinux.
Freshly installed php-fpm7. Did a `systemctl edit php-fpm` and now I am told that
`Warning: The unit file, source configuration file or drop-ins of php-fpm7.service changed on disk. Run 'systemctl daemon-reload' to reload units.`
still after doing a daemon-reload...
Any advice?
